src/core/io/writer/h5md_core.hpp

```diff
@@ -26,186 +26,185 @@
+   * @param velocity_unit The unit for velocity.
+   * @param charge_unit The unit for charge.
+   * @param comm The MPI communicator.
+   */
+  File(std::string file_path, std::string script_path, std::string mass_unit,
+       std::string length_unit, std::string time_unit, std::string force_unit,
+       std::string velocity_unit, std::string charge_unit,
+       boost::mpi::communicator comm = boost::mpi::communicator())
+      : m_file_path(std::move(file_path)),
+        m_script_path(std::move(script_path)),
+        m_mass_unit(std::move(mass_unit)),
+        m_length_unit(std::move(length_unit)),
+        m_time_unit(std::move(time_unit)), m_force_unit(std::move(force_unit)),
+        m_velocity_unit(std::move(velocity_unit)),
+        m_charge_unit(std::move(charge_unit)), m_comm(std::move(comm)) {
+    init_file();
+  };
   ~File() = default;
-  /**
-   * @brief Initialize the File object.
-   */
-  void InitFile();
+
   /**
    * @brief Method to perform the renaming of the temporary file from
    * "filename" + ".bak" to "filename".
    */
-  void Close();
+  void close();
 
-  enum WriteData {
-    W_POS = 1 << 0,
-    W_V = 1 << 1,
-    W_F = 1 << 2,
-    W_TYPE = 1 << 3,
-    W_MASS = 1 << 4,
-    W_CHARGE = 1 << 5
-  };
   /**
-   * @brief General method to write to the datasets which calls more specific
-   * write methods.
-   * Boolean values for position, velocity, force and mass.
-   */
-  void Write(int write_dat, PartCfg &partCfg, const ParticleRange &particles);
-
-  std::string &filename() { return m_filename; };
-  std::string &scriptname() { return m_scriptname; };
-  // Returns the int that describes which data should be written to the dataset.
-  int &what() { return m_what; };
-  // Returns the boolean value that describes whether data should be written to
-  // the dataset in the order of ids (possibly slower on output for many
-  // particles).
-  bool &write_ordered() { return m_write_ordered; };
-
-  // Unit system
-  bool is_unit_system_defined() {
-    return m_time_unit.length() > 0 && m_mass_unit.length() > 0 &&
-           m_length_unit.length() > 0;
-  }
-  std::string &time_unit() { return m_time_unit; };
-  std::string &mass_unit() { return m_mass_unit; };
-  std::string &length_unit() { return m_length_unit; };
-  std::string &force_unit() { return m_force_unit; };
-  std::string &velocity_unit() { return m_velocity_unit; };
-  std::string &charge_unit() { return m_charge_unit; };
+   * @brief Write data to the hdf5 file.
+   * @param particles Particle range for which to write data.
+   * @param time Simulation time.
+   * @param step Simulation step (monotonically increasing).
+
+   */
+  void write(const ParticleRange &particles, double time, int step);
+
   /**
-   * @brief Method to force flush to h5md file.
+   * @brief Retrieve the path to the hdf5 file.
+   * @return The path as a string.
    */
-  void Flush();
+  std::string &file_path() { return m_file_path; };
```

Comment:
Shouldn't this be a `const &`, or is the client supposed to change the path after the file has been created?